modules/Bio/EnsEMBL/Compara/RunnableDB/PairAligner/PairAlignerCodingExonSummary.pm

```diff
@@ -69,10 +88,17 @@ sub fetch_input {
   }
 
   my $genome_dbs = $mlss->species_set->genome_dbs;
-  my ($ref_genome_db, $non_ref_genome_db) = @$genome_dbs;
-  unless (($genome_dbs->[0]->name eq $mlss->get_value_for_tag('reference_species'))
-      && (!$mlss->has_tag('reference_component') || ($genome_dbs->[0]->genome_component eq $mlss->get_value_for_tag('reference_component')))) {
-        ($non_ref_genome_db, $ref_genome_db) = @$genome_dbs;
+  my ($ref_genome_db, $non_ref_genome_db);
+  if (scalar(@{$genome_dbs}) == 1) {
+      # Self-alignment
+      $ref_genome_db = $genome_dbs->[0];
+  } else {
+      if (($genome_dbs->[0]->name eq $mlss->get_value_for_tag('reference_species')) && (!$mlss->has_tag('reference_component')
+          || ($genome_dbs->[0]->genome_component eq $mlss->get_value_for_tag('reference_component')))) {
+          ($ref_genome_db, $non_ref_genome_db) = @$genome_dbs;
+      } else {
+          ($non_ref_genome_db, $ref_genome_db) = @$genome_dbs;
+      }
   }
   $non_ref_genome_db ||= $ref_genome_db;
```

Comment:
Have a look `MethodLinkSpeciesSet::find_pairwise_reference` as it does the same thing (but is not aware of components yet)

Reply:
I think this comment is a duplicate (unless I'm missing something).